apis/metal3.io/v1alpha1/hostfirmwaresettings_types.go

```diff
@@ -0,0 +1,169 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WriteableFirmwareSetting describes a FirmwareSetting whose value can be changed via Ironic clean-steps.
+type WriteableFirmwareSetting struct {
+	// Identifier for the Firmware setting.
+	Name string `json:"name" required:"true"`
+
+	// Value of the Firmware setting.
+	Value string `json:"value" required:"true"`
```

Comment:
Is the value always a string? I think this means that you'll be unable to apply yaml like:

```
spec:
   writableFirmwareSettings:
     someSetting: 42
```

and instead you'll have to enclose all numbers/bools in quotes. (I could be wrong, you should check.) That would be undesirable I think.

Reply:
The value is always a string, that's how Ironic it comes from Ironic - https://docs.openstack.org/api-ref/baremetal/?expanded=list-all-bios-settings-by-node-detail#node-bios-nodes, and how its gets set via clean-steps. All integers will need to be quoted.